tests/components/konnected/test_init.py

```diff
@@ -0,0 +1,601 @@
+                        },
+                    },
+                    "id": "aabbccddeeff",
+                    "host": "192.168.1.1",
+                    "port": 1234,
+                }
+            ],
+        }
+    }
+
+    # check pin to zone
+    config = {
+        konnected.DOMAIN: {
+            konnected.CONF_ACCESS_TOKEN: "abcdefgh",
+            konnected.CONF_DEVICES: [
+                {
+                    konnected.CONF_ID: "aabbccddeeff",
+                    "binary_sensors": [
+                        {"pin": 2, "type": "door"},
+                        {"zone": 1, "type": "door"},
+                    ],
+                }
+            ],
+        }
+    }
+    assert konnected.CONFIG_SCHEMA(config) == {
+        "konnected": {
+            "access_token": "abcdefgh",
+            "devices": [
+                {
+                    "default_options": {
+                        "blink": True,
+                        "discovery": True,
+                        "io": {
+                            "1": "Binary Sensor",
+                            "10": "Disabled",
+                            "11": "Disabled",
+                            "12": "Disabled",
+                            "2": "Binary Sensor",
+                            "3": "Disabled",
+                            "4": "Disabled",
+                            "5": "Disabled",
+                            "6": "Disabled",
+                            "7": "Disabled",
+                            "8": "Disabled",
+                            "9": "Disabled",
+                            "alarm1": "Disabled",
+                            "alarm2_out2": "Disabled",
+                            "out": "Disabled",
+                            "out1": "Disabled",
+                        },
+                        "binary_sensors": [
+                            {"inverse": False, "type": "door", "zone": "2"},
+                            {"inverse": False, "type": "door", "zone": "1"},
+                        ],
+                    },
+                    "id": "aabbccddeeff",
+                }
+            ],
+        }
+    }
+
+
+async def test_setup_with_no_config(hass):
+    """Test that we do not discover anything or try to set up a Konnected panel."""
+    assert await async_setup_component(hass, konnected.DOMAIN, {})
+
+    # No flows started
+    assert len(hass.config_entries.flow.async_progress()) == 0
+
+    # Nothing saved from configuration.yaml
+    assert hass.data[konnected.DOMAIN][konnected.CONF_ACCESS_TOKEN] is None
+    assert hass.data[konnected.DOMAIN][konnected.CONF_API_HOST] is None
+    assert konnected.YAML_CONFIGS not in hass.data[konnected.DOMAIN]
+
+
+async def test_setup_defined_hosts_known_auth(hass):
+    """Test we don't initiate a config entry if configured panel is known."""
+    MockConfigEntry(
+        domain="konnected",
+        unique_id="112233445566",
+        data={"host": "0.0.0.0", "id": "112233445566"},
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Possible binding to all interfaces.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4986898)